app/models/game.rb

```diff
@@ -24,6 +24,7 @@ class Game < ApplicationRecord
   
   def generate_invite_code
     self.invite_code = SecureRandom.urlsafe_base64
+    self.save
```

Comment:
If this method is going to save the record, it should be named `generate_invite_code!`, but since it's also being used in a `before_save` callback, I wouldn't save here anyway.

Maybe add a second method, `regenerate_invite_code!`, that you can use in the relevant controller. 

```ruby
def regenerate_invite_code!
  generate_invite_code
  save!
end
```

Also, try to avoid using `self.method_name` or `self.method_name`. `self` is already implied and should only be used in exceptions where it would cause issues if you left it out.